### PR TITLE
Add Endpoint for Course Entitlement Product creation

### PR DESCRIPTION
**Pull request by Mjloturco**
Adds create method for product view, enabling post for Course Entitlement products.

### CODE REVIEW

**Comment by Mjloturco** [comment]
Reviewers, this is my current approach to this. I looked at moving creation logic into the serializer, but was running into issues around nested serialization. This PR uses the create_or_update_course_entitlement util method to create a valid product instead of relying on the serializer.

comments/implementation ideas/nits welcome

File: ecommerce/extensions/api/serializers.py [hunk 1]
```diff
@@ -178,6 +179,7 @@ class ProductSerializer(ProductPaymentInfoMixin, serializers.HyperlinkedModelSer
     """ Serializer for Products. """
     attribute_values = serializers.SerializerMethodField()
     product_class = serializers.SerializerMethodField()
+    # product_class = serializers.SlugRelatedField(queryset=ProductClass.objects.all(), slug_field='name')
```
Comment (mjfrey) [inline ecommerce/extensions/api/serializers.py]: either remove or not -- do not leave commented out.

File: ecommerce/extensions/api/v2/views/courses.py [hunk 2]
```diff
@@ -4,7 +4,7 @@
 from oscar.core.loading import get_model
 from rest_framework import status
 from rest_framework.decorators import detail_route
-from rest_framework.permissions import IsAdminUser, IsAuthenticated
+from rest_framework.permissions import IsAdminUser, IsAuthenticated  # pylint: disable=unused-import
```
Comment (mjfrey) [inline ecommerce/extensions/api/v2/views/courses.py]: why the pylint pragma?

File: ecommerce/extensions/api/v2/views/products.py [hunk 3]
```diff
@@ -16,7 +19,7 @@ class ProductViewSet(NestedViewSetMixin, NonDestroyableModelViewSet):
     serializer_class = serializers.ProductSerializer
     filter_backends = (filters.DjangoFilterBackend,)
     filter_class = ProductFilter
-    permission_classes = (IsAuthenticated, IsAdminUser,)
+    # permission_classes = (IsAuthenticated, IsAdminUser,)
```
Comment (mjfrey) [inline ecommerce/extensions/api/v2/views/products.py]: remove?

File: ecommerce/extensions/api/v2/views/products.py [hunk 4]
```diff
@@ -29,3 +33,39 @@ def get_queryset(self):
...
+                product_creation_fields['price'],
+                product_creation_fields['partner'],
+                product_creation_fields['UUID'],
+                product_creation_fields['name']
+            )
+
+            data = self.serializer_class(entitlement, context={'request': request}).data
+            return Response(data, status=status.HTTP_201_CREATED)
+        else:
+            bad_rqst = "Bad Request, Product API currently only supports creation of 'Course Entitlement' products"
```
Comment (mjfrey) [inline ecommerce/extensions/api/v2/views/products.py]: I would use COURSE_ENTITLEMENT_PRODUCT_CLASS_NAME
and not just a string. And no need for the 'Bad Request'

**Review by mjfrey — APPROVED** [review]

**Review by McKenzieW — APPROVED** [review]
👍 eggs sell ent job Turc

**Pull request merged**